### ISSUE TITLE
Where should I place the svg files?

### ISSUE THREAD

**Issue by dltmd6262** [matc4/react-native-svg-uri#12]
Hello, thanks for a great library!

I've been having some trouble with svg file placement.
First I tried to store it in the same folder as images, then Android build complains that the files must be .png or .xml.
So I tried to store it in res/raw folder, but I can't access the files in the raw folder from React Native.

Where should I be storing the svg files?

**Comment by henrikra** [comment]
Hmm on iOS I have no trouble storing local images in my src/images folder

**Comment by dltmd6262** [comment]
Yeah.. apparently android wont allow anything other than some image formats and xml in the drawable folder :( im still not sure what a good workaround would be

**Comment by matc4** [comment]
Maybe a workarround for this issue is pass the prop svgXmlData to the component with the SVG File as a String.

**Comment by Crash--** [comment]
I've no issue on Android using relative path for an SVG file. 

```jsx
 <SvgUri
            style={styles.backgroundImage}
            height={122}
            width={window.width - sizes.cardMargin * 2}
            source={require('../../../../../img/landscape.svg')}
          />
```

Where img is in the same folder than index.android.js

**Comment by matc4** [comment]
@Crash-- Works in debug mode and also when you package the APK for production?

**Comment by vitorebatista** [comment]
I have the same problem... :(

**Comment by Crash--** [comment]
Had the same issue... and my workaround is... Hmm. 

I created as many React Components as SVG Files. Every React Component takes a property with the SVG content... inline. Don't know if it the best workaround or not, but it works!

**Comment by matc4** [comment]
@Crash-- I think is the only workaround for static resources in Android at the moment.

**Comment by SandroMachado** [comment]
I think this should be added to the `README.md` of the project. This is an important issue. I only notice this issue generating the release build and it took me some time to change everything to use the `svgXmlData` prop. 

Just a note in the `README.md` would help to identify this issue sooner and use the library using the `svgXmlData` prop instead of using the `source` prop.

**Comment by matc4** [comment]
@SandroMachado added to README.md

Thanks!

**Comment by slicejunk** [comment]
could it be a solution ?
https://github.com/quadric/babel-plugin-inline-import

**Comment by Naoto-Ida** [comment]
I'm having issues with iOS too. RN 0.41.0

**Comment by LeiZeng** [comment]
@slicejunk, thanks for your suggestion, I've got a working demo:
https://github.com/LeiZeng/react-native-svg-demo

**Comment by sytolk** [comment]
Thanks @LeiZeng it's works for me

**Comment by niranjan-b-prajapati** [comment]
@LeiZeng  Thank you so much for the fix. Appreciate it.

**Comment by msand** [comment]
I've got a project where the solution of @slicejunk & @LeiZeng didn't work verbatim, although I can get his example working. Seems metro-bundler ignores the .babelrc plugins configuration for some reason I haven't found yet. Switched to https://github.com/callstack/haul and now babel-plugin-inline-import works.

**Comment by adamski** [comment]
Does @LeiZeng's fix also work on iOS, or do we still need to use `require` on iOS? 

Also, if so, is there a way to prevent `--assets-dest` in the packager command from copying SVG files along with other images; so we can then access them with the same path as on iOS?

**Comment by adamski** [comment]
I'd just like to confirm that the combination of @LeiZeng's example plus [haul](https://github.com/callstack/haul) finally got this working for me on Android.

**Comment by eXhausted** [comment]
Seems this solution doesn't work for iOS? 

Got this error:
`
Warning: Failed prop type: Invalid prop `svgXmlData` of type `number` supplied to `SvgUri`, expected `string`.
`

**Comment by adamski** [comment]
@eXhausted that's a shame; I was really hoping it would, as its quite a lot of code changes for an app that is SVG heavy. I'll be attempting this soon anyhow, once I get back to the iOS version of my current project.

**Comment by charleenfei** [comment]
Hi all, has anyone else here gotten and solved the "trying to add unknown view tag" error? I also have a babel-inline-import-loader/haul setup, it seems that the svg is being imported by babel, but something goes south when this svg is then rendered by the final React Component.

**Comment by jslok** [comment]
I couldn't get it to work with babel-plugin-inline-import but it works for me on real device with https://www.npmjs.com/package/babel-plugin-inline-svg

**Comment by tastafur** [comment]
for those of us who do not use haul, I also inform that it does not work for me to modify the size of the svg but at least it works to show the svg, we should investigate how to make it compatible so that we can also pass the sizes.
and thank you very much for the contribution

https://www.npmjs.com/package/babel-plugin-inline-svg

**Comment by ArfaouiHamouda** [comment]
for those who are trying @LeiZeng  Solution i would recommend them to import file as this example
`import * as file from "./myshape.svg" `
and also to use `babel-plugin-inline-import`
